components/cdc/src/channel.rs

```diff
@@ -244,22 +252,57 @@ impl From<MemoryQuotaExceeded> for SendError {
     }
 }
 
+pub struct ObservedEvent {
+    pub created: Instant,
+    pub event: CdcEvent,
+    pub size: usize,
+}
+
+pub struct ScanedEvent {
```

Comment:
It looks to be not necessary to keep a `truncated` flag for every event. How about pass the flag to `Sink` or `Drain`.

Reply:
`Drain` is shared by many regions in one connection. If we keep it in `Drain`, a `HashMap` will be introduced.